src/org/zaproxy/zap/GuiBootstrap.java

```diff
@@ -345,48 +343,60 @@ private static void setDefaultViewLocale(Locale locale) {
      * {@link java.net.InetAddress InetAddress}) preventing some ZAP options from being correctly applied.
      */
     private void setupLookAndFeel() {
-        if (lookAndFeelSet) {
-            return;
+    	if (lookAndFeelSet) {
+    	   return;
+    	}
+    	    	
+    	if (setLookAndFeel(System.getProperty("swing.defaultlaf"))) {
+           return;
         }
-        lookAndFeelSet = true;
-
-        String lookAndFeelClassname = System.getProperty("swing.defaultlaf");
-        if (lookAndFeelClassname != null) {
-            try {
-                UIManager.setLookAndFeel(lookAndFeelClassname);
-                return;
-            } catch (final UnsupportedLookAndFeelException
-                     | ClassNotFoundException
-                     | ClassCastException
-                     | InstantiationException
-                     | IllegalAccessException e) {
-                logger.warn("Failed to set the specified look and feel: " + e.getMessage());
-            }
+    	
+    	OptionsParam options = Model.getSingleton().getOptionsParam();
+        	
+        if (setLookAndFeel(getLookAndFeelClassname(options.getViewParam().getLookAndFeel()))) {
+           return;
         }
-
-        try {
-            // Set the systems Look and Feel
-            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
-
-            if (Constant.isMacOsX()) {
-                OsXGui.setup();
-            } else {
-                // Set Nimbus LaF if available
-                for (final LookAndFeelInfo info : UIManager.getInstalledLookAndFeels()) {
-                    if ("Nimbus".equals(info.getName())) {
-                        UIManager.setLookAndFeel(info.getClassName());
-                        break;
-                    }
-                }
-            }
-        } catch (final UnsupportedLookAndFeelException
-                 | ClassNotFoundException
-                 | InstantiationException
-                 | IllegalAccessException e) {
-            logger.warn("Failed to set the \"default\" look and feel: " + e.getMessage());
+        
+        if (Constant.isMacOsX()) {
```

Comment:
Just noticed that we might need to do this earlier, before attempting to set the L&Fs (as this does other UI setup), but would need feedback from macOS users...

Reply:
In the previous code, this check is done at last so I decided to keep it at last.

Reply:
Yeah, fine to leave it as is.